include/oneapi/dpl/internal/binary_search_impl.h

```diff
@@ -75,9 +75,7 @@ lower_bound_impl(Policy&& policy, InputIterator1 start, InputIterator1 end, Inpu
                  InputIterator2 value_end, OutputIterator result, StrictWeakOrdering comp)
 {
     return oneapi::dpl::transform(policy, value_start, value_end, result,
-                                  [=](typename ::std::iterator_traits<InputIterator2>::reference val) {
-                                      return ::std::lower_bound(start, end, val, comp) - start;
-                                  });
+                                  [=](auto val) { return ::std::lower_bound(start, end, val, comp) - start; });
```

Comment:
Looks like it's not equivalent change.

Reply:
As we discussed offline, perhaps `const auto&` is most appropriate to avoid a copy.  The same for all three of these.
I double checked, and see no legitimate reason this was originally written to use a non-const lvalue ref.